RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -115,13 +193,58 @@ extension PurchaseHandler {
         }
     }
 
+    @MainActor
+    func performExternalPurchaseLogic(package: Package) async throws -> PurchaseResultData {
+        Logger.debug(Strings.executing_external_purchase_logic)
+
+        self.packageBeingPurchased = package
+        self.purchaseResult = nil
+        self.purchaseError = nil
+        self.performPurchase = PurchaseResultReporter(storeProduct: package.storeProduct,
+                                                      reportPurchaseResult: self.reportExternalPurchaseResult)
+
+        self.startAction()
+
+        return PurchaseResultData(nil, try await self.purchases.customerInfo(), false)
+    }
+
+    @MainActor
+    func reportExternalPurchaseResult(_ userCancelled: Bool, _ error: Error?) {
+        self.actionInProgress = false
+        self.performPurchase = nil
+
+        if let error {
+            self.purchaseError = error
+        } else {
+            if userCancelled {
+                self.trackCancelledPurchase()
+            } else {
+                withAnimation(Constants.defaultAnimation) {
+                    self.purchased = true
+                }
+            }
+        }
+    }
+
+    // MARK: - Restore
+
+    func restorePurchases() async throws -> (info: CustomerInfo, success: Bool) {
+        switch self.purchases.purchasesAreCompletedBy {
+        case .revenueCat:
+            return try await performRestorePurchases()
+        case .myApp:
+            return try await performExternalRestoreLogic()
+        }
```

Comment:
curious about how this works - is the SDK still getting configured in this scenario, and observing queue / delegates?

Reply:
When `purchasesAreCompletedBy` equals `.revenueCat`, there is no change in behavior. When `purchasesAreCompletedBy` equals `.myApp`, all behavior is as it was when `observerMode` was set to `true` (so I assume this means the answer to your questions about configuration/queues/delegates is "yes"). The change here, which is an addition, is that when someone taps restore on one of our paywalls, the app developer will get a callback via a view modifier that allows them to know this happened and execute their own restore code.

Does that answer everything?

Reply:
makes sense. What happens if I set it up as "purchases completed by my app" but I don't set up the view modifiers?

Reply:
You'll see a log message in the console that an external purchase/restore has started (we are discussing the exact wording needed in this PR), but if you haven't set up the view modifiers, there will be no error. I tried to think of a good way of detecting this but couldn't, short of setting up a hacky singleton. Discussed somewhere briefly with Josh, did not come to any solution for this. I'll look at this again, it would be a lot better if it actually could identify the bad state....

Reply:
honestly it feels like the way that this is _usually_ solved is that instead of "purchases completed by my app", you provide a delegate. 
If delegate exists, we call it. If not, we use our own logic. That way you can't possibly set it up wrong, right?

Reply:
Had a huddle about this with Andy, we are considering moving from view modifiers to a new initializer, to be continued next week. I'm going to separate out the API change so that these discussions don't hold up other projects.

Reply:
That makes sense to me! 💪